packages/frontend/components/Borrow/TransactionProcessingModal.tsx

```diff
@@ -0,0 +1,311 @@
+                Borrowing on Polygon Network
+              </Typography>
+            </>
+          )}
+        </Box>
+        {transactionStatus && (
+          <DialogContent>
+            <Stepper
+              activeStep={activeStep}
+              orientation="vertical"
+              connector={<CustomConnector />}
+            >
+              {steps.map((step, index) => (
+                <Step key={step.label}>
+                  <Grid
+                    container
+                    justifyContent="space-between"
+                    wrap="nowrap"
+                    alignItems="center"
+                  >
+                    <Grid item>
+                      <StepLabel StepIconComponent={CustomStepIcon}>
+                        <Typography variant="body">{step.label}</Typography>
+                        <br />
+                        <a href={step.link} target="_blank" rel="noreferrer">
+                          <Typography variant="smallDark">
+                            {step.description}
+                          </Typography>
+                          {step.link && (
+                            <LaunchIcon
+                              sx={{
+                                ml: "0.3rem",
+                                fontSize: "0.6rem",
+                                color: theme.palette.info.dark,
+                              }}
+                            />
+                          )}
+                        </a>
+                      </StepLabel>
+                    </Grid>
+                    <Grid item>
+                      {activeStep === index ? (
+                        <CircularProgress size={32} />
+                      ) : (
+                        <CheckIcon
+                          sx={{
+                            backgroundColor: theme.palette.success.dark,
+                            borderRadius: "100%",
+                            padding: "0.4rem",
+                          }}
+                          fontSize="large"
+                        />
+                      )}
+                    </Grid>
+                  </Grid>
+                </Step>
+              ))}
+            </Stepper>
+            <Card
+              variant="outlined"
+              sx={{
+                p: 0,
+                textAlign: "center",
+                mt: "2.5rem",
+                maxWidth: "27rem",
+              }}
+            >
+              <CardContent>
+                <Typography variant="small">
+                  This step takes a few minutes to process. If you close this
+                  window, your transaction will still be processed.
+                </Typography>
+              </CardContent>
+            </Card>
+          </DialogContent>
+        )}
+      </Box>
+    </Dialog>
+  )
+}
+
+function CustomStepIcon(props: StepIconProps) {
+  const { palette } = useTheme()
+
+  const icons: { [index: string]: React.ReactElement } = {
```

Comment:
You can use `Record` to  have more easy to read typing

```suggestion
  const icons: Record<string, React.ReactElement> = {
```